File: cpp_100-999/495.cc
// C++
// leetcode 495
// https://leetcode-cn.com/problems/teemo-attacking/
// 简简单单一个遍历


class Solution {
public:
    int findPoisonedDuration(vector<int>& timeSeries, int duration) {
    	int stop_time = INT_MIN;
    	int res = 0;
    	for(auto x : timeSeries) {
    		if(x >= stop_time)
    			res += duration;
    		else
    			res += duration - stop_time + x;
    		stop_time = x + duration;
    	}
    	return	res;
    }
};

File: Go/Go_100-999/495.go
// Go
// leetcode 495
// https://leetcode-cn.com/problems/teemo-attacking/

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func findPoisonedDuration(timeSeries []int, duration int) int {
	res := duration
	for i := len(timeSeries) - 2; i >= 0; i-- {
		res += min(duration, timeSeries[i + 1] - timeSeries[i])
	}
	
    return res
}